firestore/src/Connector.gs

```diff
@@ -0,0 +1,288 @@
+  if (this.enableLogging) {
+    const paramString = JSON.stringify(parameter, null, 2);
+    console.log([functionName, 'request', paramString]);
+  }
+
+  const returnObject = this[functionName](parameter);
+
+  if (this.enableLogging) {
+    const returnString = JSON.stringify(returnObject, null, 2);
+    console.log([functionName, 'response', returnString]);
+  }
+
+  return returnObject;
+};
+
+
+/** 
+ * Predefined Data Studio function for specifying connector configuration.
+ *
+ * @param {Object} request Config request parameters.
+ * @return {Object} Connector configuration to be displayed to the user.
+ */
+Connector.prototype.getConfig = function(request) {
+  const projects = this.cloud.listCloudProjects();
+  const options = [];
+  projects.forEach(function(project) {
+    options.push({
+      label: project.name + ' (' + project.projectId + ')',
+      value: project.projectId
+    });
+  });
+  const optionsSorted = options.sort(function(x, y) {
+    const xId = x.label.toLowerCase();
+    const yId = y.label.toLowerCase();
+    return xId == yId ? 0 : (xId < yId ? -1 : 1);
+  });
+    
+  return {
+    configParams: [
+      {
+        name: 'project',
+        type: 'SELECT_SINGLE',
+        displayName: 'Firebase Project',
+        helpText: 'Select a Google Cloud/Firebase project that uses Firestore.',
+        options: options
+      },
+      {
+        name: 'collection',
+        type: 'TEXTINPUT',
+        displayName: 'Collection',
+        helpText: 'Select the Firestore collection to use for this Data Source. To use multiple \
+                   collections, create additional Data Sources.'
+      },
+      {
+        type: 'INFO',
+        text: 'Firestore is a schema-less database. In order to use Firestore with Data Studio, \
+               you must supply a set of fields in the collection you want to use. \
+               Use the following format: <field_name1>:<type1>, where type is one of STRING, \
+               NUMBER, BOOLEAN, or TIMESTAMP'
+      },
+      {
+        name: 'schema',
+        type: 'TEXTAREA',
+        displayName: 'Firestore "Schema"',
+        placeholder: 'field1:STRING\nfield2:STRING\nfield3:NUMBER\nfield4:BOOLEAN'
+      },
+      {
+        name: 'numResults',
+        type: 'SELECT_SINGLE',
+        displayName: 'Max Documents',
+        helpText: 'Specifies the number of documents to read on each request. Note that each page \
+                   load will fetch *all* the documents, so setting this too high may be expensive. \
+                   Consider pre-processing in Firestore if there are many documents. Defaults to 1000.',
+        options: [
+          {label: '100', value: '100'},
+          {label: '1000', value: '1000'},
+          {label: '10000', value: '10000'},
+          {label: '100000', value: '100000'},
+        ]        
+      },
+      {
+        name: 'authEmail',
+        type: 'TEXTINPUT',
```

Comment:
Since you created this code, we have a supported way of asking for username/password (or in this case username/authKey) credentials. Please see [Authenication](https://developers.google.com/datastudio/connector/auth) for full details on how to switch this over.

Reply:
Done